force-app/main/default/classes/CourseScheduledDeleteContacts.cls

```diff
@@ -0,0 +1,101 @@
+public with sharing class CourseScheduledDeleteContacts {
+    public static void courseScheduledDeleteContacts() {
+        Date date12WeeksInPast = date.today().addDays(-84);
+
+        List<String> coursesIdsToRemoveContacts = new List<String>();
+        for (Course__c course : [
+            SELECT Id
+            FROM Course__c
+            WHERE DAY_ONLY(RegistrationToDateTime__c) = :date12WeeksInPast
+        ]) {
+            coursesIdsToRemoveContacts.add(course.Id);
+        }
+        if (!coursesIdsToRemoveContacts.isEmpty()) {
+            List<CourseRegistration__c> crList12WeeksOld = fetchCourseRegistrations(coursesIdsToRemoveContacts);
+            Set<Id> crContactIds12WeeksOld = fetchCrContactIds(coursesIdsToRemoveContacts);
+
+            Set<Id> contactsOtherThanKursdeltakere = checkAccountContactRelationship(crContactIds12WeeksOld);
+            Set<Id> sameContactIdsOnSoonerCourse = checkIfSameContactsSoonerCourses(
+                crContactIds12WeeksOld,
+                date12WeeksInPast
+            );
+
+            removePersonalDataFromCr(crList12WeeksOld);
+
+            contactsOtherThanKursdeltakere.addAll(sameContactIdsOnSoonerCourse); //Add contacts that have relationship to other Accounts than 'Kursdeltakere' or should attend other course sooner than 12 weeks
+            crContactIds12WeeksOld.removeAll(contactsOtherThanKursdeltakere); //Remove contact IDs that should not be deleted (described in line over)
+
+            deleteContacts12WeeksOld(crContactIds12WeeksOld);
```

Comment:
```suggestion
            List<CourseRegistration__c> courseRegistrationsToClean = fetchCourseRegistrationsToClean(coursesIdsToRemoveContacts);
            removePersonalDataFromCourseRegistrations(courseRegistrationsToClean);

            Set<Id> contactsToDelete = fetchContactsToDelete(coursesIdsToRemoveContacts);
            deleteContacts(contactsToDelete);
```

Et par endringer jeg vil foreslå her:

1. Du har skrevet noen kommentarer som egentlig beskriver funksjonene/metodene over, ikke `contactsOtherThanKursdeltakere.addAll` delen. Endre heller funksjonsnavnet til å være litt mer beskrivende (ref. [Clean Code boka om kommentarer](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules))
2. Endre `crList12WeeksOld` til `courseRegistrationsToClean` beskriver bedre hva det er (`cr` er litt uklart, ref. [Clean Code om variabelnavn](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#understandability-tips))
3. Endre `fetchCourseRegistrations` til `fetchCourseRegistrationsToClean`, som beskriver nøyaktig hva man får ut
4. Endre `fetchCrContactIds` til `fetchContactsToDelete` (samme som forrige punkt)
    - Flytte `checkAccountContactRelationship` inn til `fetchContactsToDelete`
    - Flytte `checkIfSameContactsSoonerCourses` inn til `fetchContactsToDelete`
    - Ettersom disse to funksjonene sitt endelig mål er å finne hvilke kontakter som kan slettes, så kan du like godt samme disse her.
    - **MERK!** `fetchContactsToDelete` burde kalle på tre funksjoner, `fetchContacts`, `checkAccountContactRelationship` og `checkIfSameContactsSoonerCourses`
5. 'deleteContacts12WeeksOld` til `deleteContacts`, ettersom det bør være tydelig allerede at man sletter 14 uker gamle kontakter.